core/shell.py

```diff
@@ -548,8 +548,28 @@ def Main(lang, arg_r, environ, login_shell, loader, readline):
   assert home_dir is not None
   rc_path = flag.rcfile
   # mycpp: rewrite of or
+
+  rc_paths = []
+  # TODO: Decide if this should be first or last rc_path (as in: should it overwrite modular includes or not)
   if rc_path is None:
-    rc_path = os_path.join(home_dir, '.config/oil/%src' % lang)
+    rc_paths.append(os_path.join(home_dir, '.config/oil/%src' % lang))
+  else:
+    rc_paths.append(rc_path)
+
+  # Experimentally try to load all files in ~/.config/oil/oshrc.d | ~/.config/oil/oilrc.d
+  # TODO: Decide on proper directory, `oil/oshrcd.d` seems ugly
```

Comment:
Yeah I agree it's a little odd, but it's consistent

I think it will be

```
~/config/oils/
  oshrc
  oshrc.d
```

`--rcfile`

`--rcdir`

?

Reply:
It was the first thing I came up with. We can also brainstorm on a better naming (for the flag and the directory) over on zulip :)